src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,499 @@
+                'template_manager' => function ($c) {
+                    $templateManager = new TemplateManager($this->eventManager, $this->eventFactory);
+                    $templateManager->registerTemplates($c->get('wp_bookings_ui/templates'));
+                    return $templateManager;
+                },
+                'assets_urls_map' => function ($c) {
+                    $assetsUrlsMap = require_once $c->get('wp_bookings_ui/assets_urls_map_path');
+                    return $this->_getContainerFactory()->make([
+                        ContainerFactoryInterface::K_DATA => $assetsUrlsMap
+                    ]);
+                }
+            ]);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function run(ContainerInterface $c = null)
+    {
+        $this->metaboxPageId = $c->get('wp_bookings_ui/metabox/post_type');
+
+        /* @var EventManagerInterface $eventManager */
+        $eventManager = $c->get('event_manager');
+
+        /** @var TemplateManager $templateManager */
+        $templateManager = $c->get('template_manager');
+
+        $assetsConfig = $c->get('assets_urls_map');
+
+        $this->_attach('admin_enqueue_scripts', function () use ($assetsConfig, $c) {
+            $this->_enqueueAssets($assetsConfig, $c);
+        }, 999);
+
+        $this->_attach('admin_init', function () use ($eventManager, $templateManager, $c) {
+            $this->_adminInit($eventManager, $templateManager, $c);
+        });
+
+        $this->_attach('admin_menu', function () use ($templateManager, $c) {
+            $this->_adminMenu($templateManager, $c);
+        });
+
+        $this->_attach('wp_ajax_set_' . $c->get('wp_bookings_ui/screen_options/key'), function () use ($c) {
+            $data = json_decode(file_get_contents('php://input'), true);
+            $statuses = $data['statuses'];
+            $this->_setScreenStatuses($c, $statuses);
+        });
+    }
+
+    /**
+     * Check current screen is screen where EDDBK UI should be rendered.
+     *
+     * @return bool
+     */
+    protected function _onAppPage()
+    {
+        return in_array(get_current_screen()->id, [
+            $this->bookingsPageId,
+            $this->metaboxPageId,
+        ]);
+    }
+
+    /**
+     * Check current screen is bookings page.
+     *
+     * @return bool
+     */
+    protected function _onBookingsPage()
+    {
+        return get_current_screen()->id === $this->bookingsPageId;
+    }
+
+    /**
+     * Get app state for booking page.
+     *
+     * @param ContainerInterface $c
+     * @return array
+     */
+    protected function _getBookingsAppState($c)
+    {
+        /* @var ControllerInterface $controller */
+        $controller = $c->get('eddbk_services_controller');
+        $services = iterator_to_array($controller->get());
+
+        return [
+            /*
+             * All available statuses in application.
+             */
+            'statuses' => $this->_trigger('eddbk_bookings_statuses', [

```

Comment:
This stuff looks like it can go into a config

Reply:
I actually have a module that provides the statuses as config.

https://github.com/RebelCode/rcmod-eddbk-booking-logic/blob/task/initial-module-code/config.php#L9

Reply:
@romalytvynenko, make sure that all logic uses a single source of truth, which is obviously the booking module.